community/kernel/src/main/java/org/neo4j/kernel/spi/legacyindex/IndexImplementation.java

```diff
@@ -107,4 +107,11 @@
      */
     @Override
     void shutdown() throws Throwable;
+
+    /**
+     * Provide store migrator for particular index implementation that participate in store migration and
+     * should be able to perform cross version migration of index implementation
+     * @return index implementation store migration
+     */
+    StoreMigrationParticipant getStoreMigrator();
```

Comment:
Provide default implementation `NOT_PARTICIPATING` to not cause problems for actual custom implementations out there?